components/script/dom/cssstylerule.rs

```diff
@@ -71,4 +76,27 @@ impl CSSStyleRuleMethods for CSSStyleRule {
             )
         })
     }
+
+    // https://drafts.csswg.org/cssom/#dom-cssstylerule-selectortext
+    fn SelectorText(&self) -> DOMString {
+        let guard = self.cssrule.shared_lock().read();
+        let stylerule = self.stylerule.read_with(&guard);
+        return DOMString::from_string(stylerule.selectors.to_css_string());
+    }
+
+    // https://drafts.csswg.org/cssom/#dom-cssstylerule-selectortext
+    fn SetSelectorText(&self, value: DOMString) {
+        // It's not clear from the spec if we should use the stylesheet's namespaces.
+        // https://github.com/w3c/csswg-drafts/issues/1511
+        let namespaces = self.cssrule.parent_stylesheet().style_stylesheet().namespaces.read();
+        let parser = SelectorParser {
+            stylesheet_origin: Origin::Author,
+            namespaces: &namespaces,
+        };
+        if let Ok(mut s) = SelectorList::parse(&parser, &mut CssParser::new(&*value)) {
+            let mut guard = self.cssrule.shared_lock().write();
+            let mut stylerule = self.stylerule.write_with(&mut guard);
+            mem::swap(&mut stylerule.selectors, &mut s);
+        }
+    }
```

Comment:
This needs to somehow invalidate stylesheets, how do changes to the rules get reflected on the page otherwise?